File: 01-programming-languages/C#/01-fundamentals/03-control-flow/conditionals.cs
// C# Control Flow - Conditionals

using System;

namespace ConditionalsDemo
{
    class Program
    {
        static void Main(string[] args)
        {
            // if-else statement
            int temperature = 25;
            
            if (temperature > 30)
            {
                Console.WriteLine("Hot weather");
            }
            else if (temperature > 20)
            {
                Console.WriteLine("Warm weather");
            }
            else if (temperature > 10)
            {
                Console.WriteLine("Cool weather");
            }
            else
            {
                Console.WriteLine("Cold weather");
            }
            
            // Nested if
            bool hasLicense = true;
            int age = 25;
            
            if (age >= 18)
            {
                if (hasLicense)
                {
                    Console.WriteLine("Can drive");
                }
                else
                {
                    Console.WriteLine("Need license");
                }
            }
            
            // Ternary operator
            string dayType = (DateTime.Now.DayOfWeek == DayOfWeek.Saturday || 
                            DateTime.Now.DayOfWeek == DayOfWeek.Sunday) 
                            ? "Weekend" : "Weekday";
            Console.WriteLine($"Today is: {dayType}");
            
            // switch statement
            string command = "start";
            
            switch (command)
            {
                case "start":
                    Console.WriteLine("Starting process");
                    break;
                case "stop":
                    Console.WriteLine("Stopping process");
                    break;
                case "pause":
                    Console.WriteLine("Pausing process");
                    break;
                default:
                    Console.WriteLine("Unknown command");
                    break;
            }
            
            // Switch with pattern matching
            object obj = 42;
            
            switch (obj)
            {
                case int i when i > 0:
                    Console.WriteLine($"Positive integer: {i}");
                    break;
                case int i:
                    Console.WriteLine($"Non-positive integer: {i}");
                    break;
                case string s:
                    Console.WriteLine($"String: {s}");
                    break;
                case null:
                    Console.WriteLine("Null value");
                    break;
            }
            
            // Switch expression (C# 8.0+)
            string season = DateTime.Now.Month switch
            {
                12 or 1 or 2 => "Winter",
                3 or 4 or 5 => "Spring",
                6 or 7 or 8 => "Summer",
                9 or 10 or 11 => "Fall",
                _ => "Unknown"
            };
            Console.WriteLine($"Current season: {season}");
            
            // Null-coalescing operator
            string? input = null;
            string output = input ?? "default value";
            Console.WriteLine($"Output: {output}");
            
            // Null-conditional operator
            string? name = null;
            int length = name?.Length ?? 0;
            Console.WriteLine($"Name length: {length}");
            
            Console.WriteLine("Conditionals example running");
        }
    }
}

File: 01-programming-languages/JavaScript/01-fundamentals/03-control-flow/conditionals.js
/*
 * JavaScript Control Flow: Conditionals
 * if/else, switch, ternary
 */

// ============================================
// if/else if/else Statement
// ============================================

const temperature = 25;

if (temperature > 35) {
    console.log("It's very hot!");
} else if (temperature > 25) {
    console.log("It's warm outside.");
} else if (temperature > 15) {
    console.log("It's mild.");
} else {
    console.log("It's cold!");
}

// ============================================
// Nested if Statements
// ============================================

const age = 20;
const hasLicense = true;

if (age >= 18) {
    if (hasLicense) {
        console.log("You can drive.");
    } else {
        console.log("You need a license.");
    }
} else {
    console.log("You're too young to drive.");
}

// ============================================
// Truthy and Falsy Values
// ============================================

// Falsy values: false, 0, "", null, undefined, NaN
// Truthy values: everything else

const values = [0, 1, "", "hello", null, undefined, NaN, false, true, [], {}];

values.forEach(val => {
    if (val) {
        console.log(`${JSON.stringify(val)} is truthy`);
    } else {
        console.log(`${JSON.stringify(val)} is falsy`);
    }
});

// ============================================
// Switch Statement
// ============================================

const dayOfWeek = "Monday";

switch (dayOfWeek) {
    case "Monday":
        console.log("Start of work week");
        break;
    case "Tuesday":
    case "Wednesday":
    case "Thursday":
        console.log("Mid-week");
        break;
    case "Friday":
        console.log("Almost weekend!");
        break;
    case "Saturday":
    case "Sunday":
        console.log("Weekend!");
        break;
    default:
        console.log("Invalid day");
}

// Switch with strict comparison
const value = "5";

switch (value) {
    case 5:
        console.log("Number 5");
        break;
    case "5":
        console.log("String 5");
        break;
    default:
        console.log("Something else");
}

// ============================================
// Ternary Operator (Conditional Expression)
// ============================================

const hour = 14;
const greeting = hour < 12 ? "Good Morning" : "Good Afternoon";
console.log(greeting);

// Nested ternary (use sparingly for readability)
const score = 85;
const grade = score >= 90 ? "A"
    : score >= 80 ? "B"
        : score >= 70 ? "C"
            : "F";
console.log(`Grade: ${grade}`);

// ============================================
// Logical Operators in Conditions
// ============================================

const isLoggedIn = true;
const isAdmin = false;

// AND - both conditions must be true
if (isLoggedIn && isAdmin) {
    console.log("Welcome, Admin!");
} else if (isLoggedIn) {
    console.log("Welcome, User!");
}

// OR - at least one condition must be true
if (isLoggedIn || isAdmin) {
    console.log("Access granted");
}

// NOT - inverts the boolean
if (!isAdmin) {
    console.log("Admin privileges not available");
}

// ============================================
// Short-circuit Evaluation
// ============================================

// AND (&&) returns first falsy or last value
const username = "Alice" && "User"; // "User"
const empty = "" && "User";          // ""

// OR (||) returns first truthy or last value
const name1 = "" || "Anonymous";     // "Anonymous"
const name2 = "Bob" || "Anonymous";  // "Bob"

console.log(`Username: ${username}, Name: ${name2}`);
